modules/phase_field/src/userobjects/PolycrystalUserObjectBase.C

```diff
@@ -200,31 +287,49 @@ PolycrystalUserObjectBase::finalize()
 void
 PolycrystalUserObjectBase::mergeSets()
 {
-  /**
-   * With initial conditions we know the grain IDs of every grain (even partial grains). We can use
-   * this information to put all mergeable features adjacent to one and other in the list so that
-   * merging is simply O(n).
-   */
-  _partial_feature_sets[0].sort();
-
-  auto it1 = _partial_feature_sets[0].begin();
-  auto it_end = _partial_feature_sets[0].end();
-  while (it1 != it_end)
+  // When working with _distribute_merge_work all of the maps will be empty except for one
+  for (MooseIndex(_maps_size) map_num = 0; map_num < _partial_feature_sets.size(); ++map_num)
   {
-    auto it2 = it1;
-    if (++it2 == it_end)
-      break;
+    /**
+     * With initial conditions we know the grain IDs of every grain (even partial grains). We can
```

Comment:
so there must be a communication step to ensure features split by processor boundaries share the same feature ID?

Reply:
No, we know the ID, because that _is_ the interface. For every point, the derived classes have to report back which grain(s) are active at that location. This means that processor boundaries, periodic boundaries, etc. don't interfere. Now what we don't know is who owns which chunks, but after the communication stages, we'll have all the pieces so we can sort and just merge adjacent chunks. 

Note: This part of the algorithm isn't new, we just had to add one more loop over map_num (merging processor num) to get everything joined.

Reply:
I suppose you didn't add the `libmesh/index_range.h` header?

Reply:
It's in the BASE CLASS